### PR TITLE
Fix: filesystem not to be writable by the defectdojo user

### PR DESCRIPTION
**Pull request by dsever**
- set directories to 755 but files to 644

Previously: 
```
drwxrwx--- 1 defectdojo root    4096 Oct  8 02:41 .
drwxr-xr-x 1 root       root    4096 Oct  8 11:04 ..
drwxrwx--- 1 defectdojo root    4096 Oct  8 11:04 dojo
-rwxrwx--- 1 defectdojo root     257 Oct  6 20:34 manage.py
drwxr-xr-x 3 defectdojo nogroup 4096 Oct  8 02:41 media
-rwxrwx--- 1 defectdojo root    1644 Oct  6 20:34 requirements.txt
drwxrwx--- 1 defectdojo root    4096 Oct  8 02:41 tests
-rwxrwx--- 1 defectdojo root     178 Oct  6 20:34 unit-tests.sh
-rwxrwx--- 1 defectdojo root    1423 Oct  6 20:34 wsgi.py
```
So user defectdojo can write to any of this file/folders, this is not good practice (related to immutability of docker containers, but explanation can be found here https://sysdig.com/blog/dockerfile-best-practices/#1-3 )

So with this PR user defectdojo (1001) can only read/execute files

```
drwxr-xr-x 1 root       root    4096 Oct 15 10:25 .
drwxr-xr-x 1 root       root    4096 Oct 15 10:25 ..
drwxr-xr-x 3 root       root    4096 Oct 15 10:25 docker
drwxr-xr-x 1 root       root    4096 Oct 15 10:21 dojo
-rwxr-xr-x 1 root       root     257 Jun 23  2020 manage.py
drwxr-xr-x 3 defectdojo nogroup 4096 Oct 15 10:28 media
-rw-r--r-- 1 root       root    1644 Oct 13 20:19 requirements.txt
drwxr-xr-x 1 root       root    4096 Oct 15 10:21 tests
-rwxr-xr-x 1 root       root     178 Aug  2 11:22 unit-tests.sh
-rw-r--r-- 1 root       root    1423 Jun 23  2020 wsgi.py
```

### CODE REVIEW

**Comment by alles-klar** [comment]
I think it is a good idea because we are hardening our system by default. For testing etc you can always run the container as root/with write access.

**Comment by valentijnscholten** [comment]
@dsever can you rebase or merge dev so it gets tested with the latest test suite changes from dev?

**Comment by alles-klar** [comment]
@dsever is it still a draft?

**Comment by dsever** [comment]
> @dsever is it still a draft?

Not anymore ;)

File: Dockerfile.django [hunk 1]
```diff
@@ -89,14 +89,12 @@ USER root
...
-    chmod 0750 -R /app/* && \
-	chmod g=u /app && \
-	chmod -R g=u /app/* && \
+    chown -R root:root /app && \
+    chmod -R u+rwX,go+rX,go-w /app && \
     mkdir /var/run/${appuser} && \
     chown ${appuser} /var/run/${appuser} && \
-	chmod g=u /var/run/${appuser}
+	  chmod g=u /var/run/${appuser} && \
+    mkdir -p media && mkdir -p media/threat && chown -R ${uid} media
```
Comment (Maffooch) [inline Dockerfile.django:97]: Wouldn't `mkdir -p media/threat be sufficient here? I believe the p flag creates all intermediary directories if they do not exist.

File: Dockerfile.django [hunk 2]
```diff
@@ -89,14 +89,12 @@ USER root
 RUN \
     adduser --system --no-create-home --disabled-password --gecos '' \
         --uid ${uid} ${appuser} && \
-    chown -R ${appuser} /app && \
-    chmod 0700 /app && \
-    chmod 0750 -R /app/* && \
-	chmod g=u /app && \
-	chmod -R g=u /app/* && \
+    chown -R root:root /app && \
```
Comment (Maffooch) [inline Dockerfile.django:92]: This should probably be kept as `appuser` so we don't get used to running everything as root

Comment (dsever) [reply Dockerfile.django:92]: @Maffooch not sure do I get your point. But intention to have it as a root is not to keep app files writable. While appuser can read/execute code

File: Dockerfile.django (hunk 1 above)
Comment (dsever) [reply Dockerfile.django:97]: Yep, it can be single mkdir

**Comment by dsever** [comment]
@Maffooch @alles-klar can we continue with this, we are having compliance issue because of this

**Comment by mtesauro** [comment]
I don't have a problem with this.  It makes sense.

If you really need to get into the container and mess around, you can always add `-u 0` to your `docker exec -it` and go in as root inside the container.

File: Dockerfile.django [hunk 3]
```diff
@@ -89,14 +89,12 @@ USER root
...
+    chmod -R u+rwX,go+rX,go-w /app && \
     mkdir /var/run/${appuser} && \
     chown ${appuser} /var/run/${appuser} && \
-	chmod g=u /var/run/${appuser}
+	  chmod g=u /var/run/${appuser} && \
+    mkdir -p media/threat && chown -R ${uid} media
```
Comment (mtesauro) [inline Dockerfile.django:97]: FWIW, I _believe_ that the threat directory under media is no longer used but I'd still leave it in for now as one empty directory isn't worth messing with and this dockerfile can still build older versions that need ./media/threat

**Review by mtesauro — APPROVED** [review]
Approved.

**Review by alles-klar — APPROVED** [review]

**Pull request merged**